### PR TITLE
Do not update packages if no_update is set.

### PR DESCRIPTION
**Pull request by meiyopeng**
Do not update packages if no_update is set.

### CODE REVIEW

**Comment by lotem** [comment]
This change is incorrect.
`update-package.sh` has to be called to checkout the required branch.

File: scripts/install-packages.sh [hunk 1]
```diff
@@ -57,10 +57,10 @@ fetch_or_update_package() {
     else
         if [[ -z "${option_no_update}" ]]; then
             echo $(info 'Updating package:') $(highlight "${package}")
+            "${script_dir}"/update-package.sh "${package_dir}" "${branch}"
         else
             echo $(info 'Found package:') $(highlight "${package}")
         fi
-        "${script_dir}"/update-package.sh "${package_dir}" "${branch}"
```
Comment (lotem) [inline scripts/install-packages.sh]: `no_update` is also read in this script.

File: scripts/install-packages.sh [hunk 2]
```diff
@@ -57,10 +57,10 @@ fetch_or_update_package() {
     else
         if [[ -z "${option_no_update}" ]]; then
             echo $(info 'Updating package:') $(highlight "${package}")
+            "${script_dir}"/update-package.sh "${package_dir}" "${branch}"
```
Comment (lotem) [inline scripts/install-packages.sh]: skipping this call could leave the working copy on a different branch.

**Comment by lotem** [comment]
I understand your requirement.
With `no_update=1` the script won't pull from remote unless the requested branch doesn't exist locally. You do not need this code change.

**Comment by meiyopeng** [comment]
Currently plum throw a lot of warnings because it cannot find the git command. Guix building environment is limited to only necessary dependencies, so git is not available. Could this be fixed?

**Comment by lotem** [comment]
Does the build system support "dev/build dependencies"?
If you aim to package data, git sounds like a build time dependency but not required at runtime.

**Comment by meiyopeng** [comment]
Yes I can pass git as a build time dependency, but plum still throw warnings because schema packages are not git repositories.
`WARNING: not a git repository, skipped updating './package/rime/bopomofo'`

**Comment by meiyopeng** [comment]
I am porting Rime to Guix. If you are interested, please have a look at the package definition I wrote for rime-data:
https://github.com/meiyopeng/guix/blob/rime/gnu/packages/ibus.scm#L322

**Comment by lotem** [comment]
Wocow, it's written in lisp.
There is a problem for you: the build itself isn't "reproducible", and source code is not tagged:
rime/plum doesn't yet support "locking" the version of downloaded packages, neither by tags, submodule reference, nor by saving the git ref in a lock file.

**Comment by meiyopeng** [comment]
> There is a problem for you: the build itself isn't "reproducible", and source code is not tagged:

Yeah, I know. I reported the [reproducibility issue](https://github.com/rime/librime/issues/226) to librime, but still have no answer.

> rime/plum doesn't yet support "locking" the version of downloaded packages, neither by tags, submodule reference, nor by saving the git ref in a lock file.

My solution is to manually specify commit id for every schema package and copy them into "package/rime" directory. I have already successfully built rime on Guix. I made this pull request to suppress build process warnings when building rime-data. It would be great if you can provide an option to prevent plum from trying to download packages and check out branches. The same option will also be useful to NixOS, a similar OS to Guix.

**Pull request closed without merging**